src/crypto/crypto_common.cc

```diff
@@ -480,8 +480,17 @@ MaybeLocal<Object> GetLastIssuedCert(
       return MaybeLocal<Object>();
     issuer_chain = ca_info;
 
+    // Take the value of cert->get() before and after the call to cert->reset()
+    // in order to compare them and provide a way to exit this loop
+    // in case it gets stuck
+    X509* value_before_reset = cert->get();
+
     // Delete previous cert and continue aggregating issuers.
     cert->reset(ca);
+
+    X509* value_after_reset = cert->get();
+    if (value_before_reset == value_after_reset)
```

Comment:
```suggestion
    if (value_before_reset == ca)
```

or am I missing something here?

Reply:
It would also be possible to move the condition up **above** the `cert->reset()` and just do 
```
if (cert->get() == ca)
  break;
```
and have a even more compact solution. Does anyone see a problem with that?

Reply:
It seems fine to me (although I’m not really sure that I know where the right place is either way :))

Reply:
Nevermind, that created a memory leak (for reference, see https://github.com/nodejs/node/actions/runs/705770876). Going with your suggestion then 👍 .